docs/rfc/rfc-19-random-beacon-v2.adoc

```diff
@@ -0,0 +1,157 @@
+:toc: macro
+
+= RFC 19: Random Beacon v2
+
+:icons: font
+:numbered:
+toc::[]
+
+== Background
+
+The last release of the random beacon got its genesis Sep 16th 2020 and worked
+until November 11th when it was stopped with a panic button as a result of an
+unannounced https://github.com/keep-network/keep-core/blob/main/docs/status-reports/2020-11-11-retro-geth-hardfork.adoc[Ethereum
+hard fork].
+
+For the time it was working, the beacon produced more than 1700 relay entries
+and 70 groups.
+
+Both on-chain and off-chain components proved to be successful but - just like
+with any other product - some pain points were identified by the users. This RFC
+aims at describing the pain points and proposing changes that should make the
+beacon easier to operate and more consistent with tBTC v2.
+
+=== ETH-based pricing
+
+Having reimbursements for various on-chain actions in ETH was a nice feature but
+also complex in practice. Highly volatile gas price led to creating new groups
+too often and gas price ceiling value was impossible to maintain up-to-date.
```

Comment:
why did the gas price volatility lead to creating new groups?

Reply:
Please see https://github.com/keep-network/keep-core/pull/2585/commits/ce809c28d2dea1e5edb21493021560b14b718084. This is related to the pricing scheme. Because we did not have a gas price oracle that could not be manipulated, we decided to use a governable gas price ceiling. However, the gas price over the course of a week could jump from 20 Gwei to 200 Gwei, and some fees calculated based on the ceiling were just too high. As a result, some pools, like a pool for group creation were getting filled too soon and groups were getting created too often.

Looking how gas prices are volatile today I think we should decouple a frequency of system events with the gas price, wherever possible.